src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -894,7 +902,7 @@ variableDeclarator![AST mods, AST t]
 
 declaratorBrackets[AST typ]
     :    {#declaratorBrackets=typ;}
-        (lb:LBRACK^ {#lb.setType(ARRAY_DECLARATOR);} RBRACK)*
+        ((annotation)? lb:LBRACK^ {#lb.setType(ARRAY_DECLARATOR);} RBRACK)*
```

Comment:
reference: [`VariableDeclaratorId`](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-VariableDeclaratorId), in particular `Dims` which allows an optional annotation in each []
```
VariableDeclaratorId:
  Identifier [Dims]

Dims:
  {Annotation} [ ] {{Annotation} [ ]}
```